### PR TITLE
feat: support LDAP

### PR DESCRIPTION
**Pull request by WindSpiritSR**
Fix: https://github.com/casbin/casdoor/issues/139

Sync users from LDAP server

Deployed at: https://door.leviatan.cn

Admin: `admin`
Password: `123`

LDAP server deployed at: leviatan.cn:389
phpLDAPadmin deployed at: https://pla.leviatan.cn
Admin: `cn=admin,dc=leviatan,dc=cn`
Password: `leviatanpasswd`

Signed-off-by: WindSpiritSR <simon343riley@gmail.com>

### CODE REVIEW

**Comment by hsluoyz** [comment]
@Kininaru @sh1luo plz review.

**Review by Kininaru — APPROVED** [review]

**Comment by Kininaru** [comment]
Casbin Licenses.

**Comment by hsluoyz** [comment]
@WindSpiritSR add LDAP docs to: https://github.com/casdoor/casdoor-website

**Comment by hsluoyz** [comment]
@WindSpiritSR squash your commits into one.

File: object/ldap.go [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+	conn, err := goldap.Dial("tcp", fmt.Sprintf("%s:%d", host, port))
+	if err != nil {
+		return nil, err
+	}
+
+	err = conn.Bind(adminUser, adminPasswd)
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("fail to login Ldap server with [%s]", adminUser))
+	}
+
```
[the remark this answers is not in the document]
Comment (WindSpiritSR) [reply object/ldap.go]: Fixed

**Comment by hsluoyz** [comment]
@WindSpiritSR 

add URL in Casdoor dashboard to point to the LDAP server URL like https://pla.leviatan.cn/

**Comment by WindSpiritSR** [comment]
@hsluoyz Moved ldap list to organization page.
And LDAP server does not necessarily have a web UI. phpLDAPadmin is an optional third-party servive.

**Comment by hsluoyz** [comment]
@WindSpiritSR cannot change name:

image

cannot create new:

image

**Comment by WindSpiritSR** [comment]
@hsluoyz fixed

**Comment by hsluoyz** [comment]
@WindSpiritSR 

image

**Comment by WindSpiritSR** [comment]
@hsluoyz Resolved

**Pull request merged**